public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -498,42 +495,13 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
     return (row && row.searchWords && row.searchWords.length > 0) === true;
   }
 
-  throwUnless = (err: any, condition: boolean, target: LokiQuery) => {
+  throwUnless(err: FetchError, condition: boolean, target: LokiQuery) {
     if (condition) {
       return of(err);
     }
 
-    const error: DataQueryError = this.processError(err, target);
-    throw error;
-  };
-
-  processError = (err: any, target: LokiQuery): DataQueryError => {
-    const error: DataQueryError = {
-      message: (err && err.statusText) || 'Unknown error during query transaction. Please check JS console logs.',
-      refId: target.refId,
-    };
-
-    if (err.data) {
-      if (typeof err.data === 'string') {
-        if (err.data.includes('escape') && target.expr.includes('\\')) {
-          error.message = `Error: ${err.data}. Make sure that all special characters are escaped with \\. For more information on escaping of special characters visit LogQL documentation at https://github.com/grafana/loki/blob/master/docs/logql.md.`;
```

Comment:
We actually added this message 4 days ago https://github.com/grafana/grafana/pull/26136. Not sure if it makes sense to remove all of this code. cc @ivanahuckova.

Reply:
err.data is never a string here, backendSrv transforms it (same was true before my backendSrv refactor as far as I can tell) 

https://github.com/grafana/grafana/blob/master/public/app/core/services/backend_srv.ts#L248

Reply:
but please add this back if that is the case, but check err.data.message  instead

Reply:
@ivanahuckova can you check this change and see if we should add that message somewhere else?

Reply:
yeah, I will look at it the soonest I can (after Elastic PR).

Reply:
I hope it is okay, I have added a commit to this PR that keeps the custom escape error processing https://github.com/grafana/grafana/pull/26291/commits/b428052e423f84321f5ccd3b1dfb44bc0949582e. If not, I can still revert and create separate PR.

Reply:
I have also fixed the failing test 466b01b.